core/src/test/java/org/apache/iceberg/jdbc/TestJdbcCatalog.java

```diff
@@ -167,6 +168,67 @@ public void testInitialize() {
     jdbcCatalog.initialize("test_jdbc_catalog", properties);
   }
 
+  @Test
+  public void testRetryingErrorCodesProperty() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(CatalogProperties.WAREHOUSE_LOCATION, this.tableDir.toAbsolutePath().toString());
+    properties.put(CatalogProperties.URI, "jdbc:sqlite:file::memory:?icebergDB");
+    properties.put(JdbcUtil.RETRYABLE_STATUS_CODES, "57000,57P03,57P04");
```

Comment:
we might want to double-check here in case users pass this with a space after the comma

Reply:
Updated to handle allowing users passing in a space in the comma delimited list (we just strip the spaces)